testplan/testing/base.py

```diff
@@ -867,13 +874,97 @@ def _record_driver_timing(
         case_report.extend(case_result.serialized_entries)
         case_report.attachments.extend(case_result.attachments)
 
+    def _record_driver_connection(self, case_report: TestCaseReport) -> None:
+        case_result = self.cfg.result(
+            stdout_style=self.stdout_style, _scratch=self.scratch
+        )
+        connections: List[BaseDriverConnection] = []
```

Comment:
Here is where I think a DriverConnections class is needed to aggregate the result of calling each drivers connection extractor.